src/corelibs/U2View/src/ov_assembly/AssemblyBrowserFactory.cpp

```diff
@@ -79,7 +80,31 @@ Task* AssemblyBrowserFactory::createViewTask(const MultiGSelection& multiSelecti
     }
 
     if (!asmObjects.isEmpty()) {
-        foreach (GObject* o, asmObjects) {
+        // Move up to MAX_VIEWS assembly objects with reads to the start of the list.
+        // So they will be selected to be opened with a higher priority than empty assemblies.
+
+        // Select assembly objects with reads first.
+        QList<GObject*> assemblyObjectsWithReads;
+        for (GObject* o : qAsConst(asmObjects)) {
+            if (auto assemblyObject = qobject_cast<AssemblyObject*>(o)) {
+                U2OpStatusImpl os;
+                if (assemblyObject->getReadCount(os) > 0) {
```

Comment:
You need to check the `os` status

Reply:
This check is not needed because we do not want to interrupt the whole process because of the local optimization: if read count is not available for any reason here we continue as no-op.